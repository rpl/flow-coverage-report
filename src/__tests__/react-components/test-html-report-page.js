'use babel';

/* eslint-disable react/jsx-filename-extension, import/no-dynamic-require */

import React from 'react';
import renderer from 'react-test-renderer';

import {FLOW_COVERAGE_SUMMARY_DATA} from '../fixtures';

import {BASE_DIR} from './common';

const REACT_COMPONENT = `${BASE_DIR}/html-report-page`;

test('<HTMLReportSummaryPage/>', () => {
  const HTMLReportSummaryPage = require(REACT_COMPONENT).HTMLReportSummaryPage;
  const props = {
    coverageSummaryData: {...FLOW_COVERAGE_SUMMARY_DATA, threshold: 40}
  };
  const tree = renderer.create(<HTMLReportSummaryPage {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test.skip('<HTMLReportSourceFilePage/>');
