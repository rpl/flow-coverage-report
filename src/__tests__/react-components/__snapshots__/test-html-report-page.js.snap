// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HTMLReportSummaryPage/> 1`] = `
<html>
  <head>
    <meta
      charSet="utf-8"
    />
  </head>
  <body>
    <div
      className="ui grid container"
    >
      <div
        className="row"
      >
        <h2>
          Flow Coverage Report
        </h2>
      </div>
      <div
        className="row"
      >
        <h4
          className="ui header"
        >
          Summary
        </h4>
        <table
          className="ui small celled table"
        >
          <thead>
            <tr>
              <th>
                Percent
              </th>
              <th>
                Total
              </th>
              <th>
                Covered
              </th>
              <th>
                Uncovered
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="positive"
            >
              <td>
                50
                 %
              </td>
              <td>
                10
              </td>
              <td>
                5
              </td>
              <td>
                5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="row"
      >
        <h4
          className="ui header"
        >
          Files
        </h4>
        <table
          className="ui small celled table sortable"
          id="files"
        >
          <thead>
            <tr>
              <th
                data-sort-default={true}
              >
                Filename
              </th>
              <th>
                Annotation
              </th>
              <th>
                Percent
              </th>
              <th>
                Total
              </th>
              <th>
                Covered
              </th>
              <th>
                Uncovered
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="positive"
            >
              <td
                className="selectable"
              >
                <a
                  href="sourcefiles/src/a.js.html"
                >
                  src/a.js
                </a>
              </td>
              <td>
                 @
                flow
                 
              </td>
              <td
                className=""
              >
                <span>
                  50
                   %
                </span>
              </td>
              <td>
                 
                2
                 
              </td>
              <td>
                 
                1
                 
              </td>
              <td>
                 
                1
                 
              </td>
            </tr>
            <tr
              className="positive"
            >
              <td
                className="selectable"
              >
                <a
                  href="sourcefiles/src/b.js.html"
                >
                  src/b.js
                </a>
              </td>
              <td>
                 @
                flow
                 
              </td>
              <td
                className=""
              >
                <span>
                  50
                   %
                </span>
              </td>
              <td>
                 
                2
                 
              </td>
              <td>
                 
                1
                 
              </td>
              <td>
                 
                1
                 
              </td>
            </tr>
            <tr
              className="negative"
            >
              <td
                className="selectable"
              >
                <a
                  href="sourcefiles/src/c.js.html"
                >
                  src/c.js
                </a>
              </td>
              <td>
                 @
                no flow
                 
              </td>
              <td
                className=""
              >
                <span>
                  50
                   %
                </span>
              </td>
              <td>
                 
                2
                 
              </td>
              <td>
                 
                1
                 
              </td>
              <td>
                 
                1
                 
              </td>
            </tr>
            <tr
              className="positive"
            >
              <td
                className="selectable"
              >
                <a
                  href="sourcefiles/src/d1/d.js.html"
                >
                  src/d1/d.js
                </a>
              </td>
              <td>
                 @
                flow strict
                 
              </td>
              <td
                className=""
              >
                <span>
                  50
                   %
                </span>
              </td>
              <td>
                 
                2
                 
              </td>
              <td>
                 
                1
                 
              </td>
              <td>
                 
                1
                 
              </td>
            </tr>
            <tr
              className="positive"
            >
              <td
                className="selectable"
              >
                <a
                  href="sourcefiles/src/d1/e.js.html"
                >
                  src/d1/e.js
                </a>
              </td>
              <td>
                 @
                flow strict-local
                 
              </td>
              <td
                className=""
              >
                <span>
                  50
                   %
                </span>
              </td>
              <td>
                 
                2
                 
              </td>
              <td>
                 
                1
                 
              </td>
              <td>
                 
                1
                 
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="row centered"
      >
        <footer>
          Flow Coverage Report generated by
          <a
            href="https://flowtype.org"
          >
             flow 
          </a>
          and
          <a
            href="https://github.com/rpl/flow-coverage-report"
          >
             flow-coverage-report 
          </a>
          at 
        </footer>
      </div>
    </div>
  </body>
</html>
`;
